### PR TITLE
Add test cases

### PR DESCRIPTION
**Pull request by ritvik-knagaram**
Review test cases

### CODE REVIEW

File: py-lexer-testing2.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/*
+INT_WORD
+IDENT(main)
+LPAREN
+RPAREN
+LBRACK
+INT_WORD
+IDENT(a)
+EQUALS
+INT
```
Comment (adityaxdiwakar) [inline py-lexer-testing2.c]:
```suggestion
INT(5)
```

File: py-lexer-testing2.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+SEMI
+FLOAT_WORD
+IDENT(b)
+EQUALS
+FLOAT
```
Comment (adityaxdiwakar) [inline py-lexer-testing2.c]:
```suggestion
FLOAT(6.7)
```

File: py-lexer-testing2.c [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+SEMI
+WHILE
+LPAREN
+IDENT(a)
+LT
+INT
```
Comment (adityaxdiwakar) [inline py-lexer-testing2.c]:
```suggestion
INT(10)
```

File: py-lexer-testing2.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+RPAREN
+LBRACK
+IDENT(a)
+PLUSEQ
```
Comment (adityaxdiwakar) [inline py-lexer-testing2.c]: Hm, you have `a++` written, but this is `a += INT`? I don't think we support `++` anyways, so let's do `a += 3` or something and that should lex to
```
IDENT(a)
PLUS
EQUALS
INT(3)
```
Note, that it should lex to PLUS and EQUALS as separate tokens, not PLUSEQ (as a merged token). Can you also remove the useless PLUSEQ and MINEQ from `lex.py`?

File: py-lexer-testing3.c
Comment (adityaxdiwakar) [inline py-lexer-testing3.c:1]: This file is empty.

File: py-lexer-testing4.c
Comment (adityaxdiwakar) [inline py-lexer-testing4.c:1]: This file is empty.

**Review by adityaxdiwakar — CHANGES_REQUESTED** [review]
Overall great job! Some general comments.

1. Rename test files to `test1.c`, `test2.c`, etc. These tests will also be used for `cpp-lexer`.
2. Move the test files into a `tests/` folder.
3. Let's think of some more edge cases. For example, add some tests with things like `a+=3` and `inta=b` which should lex as `IDENT(a) PLUS EQUALS INT(3)` and `IDENT(inta) EQUALS IDENT(b)`.


Thank you!

**Review by CC-734 — APPROVED** [review]

**Comment by adityaxdiwakar** [comment]
Also, please put the expected tokens into a different file.

For example:

`tests/test1.c`:
```
int main() { return 0; }
```

`tests/outputs/test1.expected`:
```
INT_WORD
IDENT(main)
LPAREN
RPAREN
LBRACK
RETURN
INT(0)
SEMI
RBRACK
```

This way, we can use @CC-734's work on Github Actions to automatically run tests by simply diffing the `py-lexer` output and the expected output from these files. Thank you.

**Comment by adityaxdiwakar** [comment]
I fixed some things. I think this is ready for merge.

**Review by adityaxdiwakar — APPROVED** [review]
